server/graphql/types/src/types/item.rs

```diff
@@ -133,6 +135,19 @@ impl ItemNode {
         Ok(result)
     }
 
+    // TODO: maybe do this in a loader!
```

Comment:
Loader does sound useful. I think we need to think about how we can split out our types though, as in some cases you won't want to load all the extra stuff in on item (if you just need name and code for example). I'm  not sure we have a good pattern established for that?

Reply:
What do you mean? If we use the loader then these wouldn't get loaded if only selecting a subset of fields..

Unless you're saying we should add another query to get the variants directly. Which probably, though I think the current cases we have, we'd only be querying the variants in the context of an item 🤷‍♀️

Reply:
I take it back this is already being done.
I see in Item API we have 

```
fragment ItemRow on ItemNode {
  __typename
  id
  code
  name
  unitName
}

fragment ItemWithPackSize on ItemNode {
  ...ItemRow
  defaultPackSize
}
fragment ItemStockOnHand on ItemNode {
  ...ItemWithPackSize
  availableStockOnHand(storeId: $storeId)
}
fragment ItemRowWithStats on ItemNode {
  ...ItemStockOnHand
  stats(storeId: $storeId) {
    __typename
    averageMonthlyConsumption
    availableStockOnHand
    availableMonthsOfStockOnHand
    totalConsumption
  }
}
```

So I think we already have what I was thinking of.
Just didn't want the loader running unless we actually need it in the context...